presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -101,6 +101,11 @@ public InformationSchemaPageSource(
         table = tableHandle.getTable();
         prefixIterator = Suppliers.memoize(() -> {
             Set<QualifiedTablePrefix> prefixes = tableHandle.getPrefixes();
+            if (!tableHandle.getLimit().isPresent()) {
+                // no limit is used, therefore it doesn't make sense to split information schema query into smaller ones
```

Comment:
Listing schemas explicitly here also helps return results to client sooner. And with respect to Hive Connector, it doesn't really increase the number of metadata calls to HMS. Having said that, I think it's fine to make the change as it may help in case of other connectors.

Reply:
> And with respect to Hive Connector, it doesn't really increase the number of metadata calls to HMS. Having said that, I think it's fine to make the change as it may help in case of other connectors.

JDBC connectors can take advantage of single query. For example, they can query their own `information_schema`, so instead of X requests you get one